great_expectations/datasource/fluent/config.py

```diff
@@ -65,85 +70,127 @@
+        _DATA_ASSET_NAME_KEY,  # The "name" field is set in validation upon deserialization from configuration key; hence, it should not be serialized.
     }
 
+    class Config:
+        extra = Extra.ignore  # ignore any old style config keys
+
     @property
-    def datasources(self) -> Dict[str, Datasource]:
+    def datasources(self) -> List[Datasource]:
+        """Returns available Fluent Datasources as list."""
         return self.fluent_datasources
 
-    class Config:
-        extra = Extra.ignore  # ignore any old style config keys
+    def get_datasources_as_dict(self) -> Dict[str, Datasource]:
+        """Returns available Datasource objects as dictionary, with corresponding name as key.
+
+        Returns:
+            Dictionary of "Datasource" objects with "name" attribute serving as key.
+        """
+        datasource: Datasource
+        datasources_as_dict: Dict[str, Datasource] = {
+            datasource.name: datasource for datasource in self.fluent_datasources
+        }
+
+        # Since dictionary keys are unique, this ensures uniqueness of "Datasource" objects.
+        self.fluent_datasources = list(datasources_as_dict.values())
+
+        return datasources_as_dict
+
+    def get_datasource_names(self) -> Set[str]:
+        """Returns the set of available Datasource names.
+
+        Returns:
+            Set of available Datasource names.
+        """
+        datasource: Datasource
+        return {datasource.name for datasource in self.datasources}
+
+    def get_datasource(self, datasource_name: str) -> Datasource:
+        """Returns the Datasource referred to by datasource_name
+
+        Args:
+            datasource_name: name of Datasource sought.
+
+        Returns:
+            Datasource -- if named "Datasource" objects exists; otherwise, exception is raised.
+        """
+        try:
+            datasource: Datasource
+            return list(
+                filter(
+                    lambda datasource: datasource.name == datasource_name,
+                    self.datasources,
+                )
+            )[0]
+        except IndexError as exc:
+            raise LookupError(
+                f"'{datasource_name}' not found. Available datasources are {self.get_datasource_names()}"
+            ) from exc
+
+    def update_datasources(self, datasources: Dict[str, Datasource]) -> None:
+        """
+        Updates internal list of datasources using supplied datasources dictionary.
+
+        Args:
+            datasources: Dictionary of datasources to use to update internal datasources.
+        """
+        datasources_as_dict: Dict[str, Datasource] = self.get_datasources_as_dict()
+        datasources_as_dict.update(datasources)
+        # Since dictionary keys are unique, this ensures uniqueness of "Datasource" objects.
+        self.fluent_datasources = list(datasources_as_dict.values())
```

Comment:
This is a clever way of finding the right datasources to update.

Reply:
@Kilo59 I tried using `set()` first, but our `Datasource` and `DataAsset` structures are non-hashable, so this did not work (that is all right -- I do not think that we should rush to implement dunder hash as using dictionary works well).  Thanks.